internal/nix/nix-darwin/multi.nix

```diff
@@ -0,0 +1,166 @@
+      suffix = if name == "" then "" else "-${name}";
+      user = topConfig.users.users.${config.user}; 
+    in
+    {
+      options = {
+        systemConfig = lib.mkOption {
+          internal = true;
+          type = types.unspecified; # A function from module arguments to config.
+        };
+        package = mkOption {
+          description = ''
+            Package containing the bin/hercules-ci-agent executable.
+          '';
+          type = types.package;
+          default = pkgs.hercules-ci-agent;
+          defaultText = literalExpression "pkgs.hercules-ci-agent";
+        };
+        user = mkOption {
+          type = types.str;
+          default = "_hercules-ci-agent";
+        };
+        group = mkOption {
+          type = types.str;
+          default = "_hercules-ci-agent";
+        };
+        logFile = mkOption {
+          type = types.str;
+          default = "/var/log/hci-agent${suffix}.log";
+        };
+        baseDirectory = mkOption {
+          type = types.str;
+          default = "/var/lib/hercules-ci-agent${if name == "" then "" else "/${name}"}"; # Almost a `suffix` logic, but yield subdir
+        };
+      } // makeSettingsOptions { cfg = config; opt = options; };
+      config = let cfg = config; in
+        {
+          settings = {
+            _module.args = {
+              packageOption = options.package;
+              inherit pkgs;
+            };
+            baseDirectory = cfg.baseDirectory; 
+            nixUserIsTrusted = true;
+            labels =
+              let
+                mkIfNotNull = x: mkIf (x != null) x;
+              in
+              {
+                darwin.label = topConfig.system.darwinLabel;
+                darwin.revision = topConfig.system.darwinRevision;
+                darwin.version = topConfig.system.darwinVersion;
+                darwin.nix.daemon = topConfig.nix.useDaemon;
+                darwin.nix.sandbox = topConfig.nix.settings.sandbox;
+              };
+          };
+          systemConfig = { config, ... }: {
+            launchd.daemons."hci-agent${suffix}" = {
+              script = "exec ${command}";
+
+              path = [ config.nix.package ];
+              environment = {
+                NIX_SSL_CERT_FILE = "${pkgs.cacert}/etc/ssl/certs/ca-bundle.crt";
+              };
+
+              serviceConfig.KeepAlive = true;
+              serviceConfig.RunAtLoad = true;
+              serviceConfig.StandardErrorPath = cfg.logFile;
+              serviceConfig.StandardOutPath = cfg.logFile;
+              serviceConfig.GroupName = cfg.user;
+              serviceConfig.UserName = cfg.group;
+              serviceConfig.WorkingDirectory = cfg.baseDirectory;
+              serviceConfig.WatchPaths = [
+                cfg.settings.staticSecretsDirectory
+              ];
+            };
+      
+            nix.settings.trusted-users = [ cfg.user ];
+
+            system.activationScripts.preActivation.text = ''
```

Comment:
I would suggest to use a different name that refers to `hercules-ci` for the activation script.

Reply:
`preActivation` is a proper name for nix-darwin (it have set of predefined names, which later substituted, I can't use arbitrary name -- https://github.com/LnL7/nix-darwin/blob/master/modules/system/activation-scripts.nix#L53 )

Reply:
Resolve it if all ok. I rebased/squashed commits